server/plugin.go

```diff
@@ -757,3 +791,49 @@ func (p *Plugin) runMetricsUpdaterTask(store store.Store, updateMetricsTaskFrequ
 	metricsUpdater()
 	model.CreateRecurringTask("metricsUpdater", metricsUpdater, updateMetricsTaskFrequency)
 }
+
+func (p *Plugin) OnSharedChannelsPing(rc *model.RemoteCluster) bool {
+	p.API.LogDebug("Ping received", "remote", rc.DisplayName, "pluginID", pluginID)
+	return true
+}
+
+func (p *Plugin) OnSharedChannelsAttachmentSyncMsg(fi *model.FileInfo, post *model.Post, rc *model.RemoteCluster) error {
+	p.API.LogDebug("SyncMsg attachment", "fileInfo", fi, "post", post, "remote", rc.DisplayName, "pluginID", pluginID)
+	if !p.configuration.MetricsForSharedChannelsInfrastructure {
+		return nil
+	}
+	p.GetMetrics().ObserveMessageSharedChannelsEvent("attachment_created")
```

Comment:
Wondering what the point of the "feature flag" here is? Feels like we should be turning it on or off depending on the feature flag, and always measuring metrics. If we can't reliably "turn it off", maybe the feature flag is itself not useful?

Reply:
We are deciding to execute or not the metrics based on the config setting, but as you said, if there is not messages flowing because the shred channels features is off that maybe is not needed.